database_connection.py

```diff
@@ -0,0 +1,131 @@
+# import json
+import os
+import psycopg2
+
+from player import Player
+
+PASSWORD = os.environ.get('PASSWORD')  # env variable
+PLAYER_TABLE = "playerinfo"  # table name
+
+
+# make sure table exists, if not create it
+def setup(conn):
```

Comment:
We should make this a general connection function instead of one that is specific to the `PLAYER_TABLE`, we could make a few minor adjustments:
```
create_commands = {
    "playerinfo": """
        CREATE TABLE playerinfo (
            player_id SERIAL PRIMARY KEY,
            name TEXT NOT NULL UNIQUE,
            games_played INT NOT NULL,
            elo INT NOT NULL,
            roles TEXT[] NOT NULL
    )"""
}

def setup(conn, table: str, command: str, create_command):
    if command is None:
        command = f"SELECT * FROM {table}"
     
    cur = conn.cursor()
    try:
        # check if table exists
        cur.execute(command)
        print(table + " table found!")
    except (Exception, psycopg2.DatabaseError):
        print(table + " table not created, creating it now...")
        # create table
        create_command = create_commands[table]
        if create_command is None:
            print("No database creation command found.")
            return
        conn.rollback()  # used to refresh after first query failed
        cur.execute(create_command)
        cur.close()
        conn.commit()
        print(table + " table created!")
```